### DISCUSSION TITLE
move `minio` executable to `/home/minio-user/bin`

### DISCUSSION THREAD

**Discussion by cgs-labs** [minio/minio#18587] [Show and tell]
I am new to minio and followed the docs. This is on Debian 12 with systemd

## Current State

- `minio` gets installed via a debian package and will be stored in `/usr/local/bin/`
- `mc` get installed separetetly via curl for the local user
- created `minio-user` user and group to get access to the drive, and to run the systemd service
- systemd service file gets installed as in the docs: running `/usr/local/bin/minio` as `minio-user`

###  `mc admin update <server>` does not have sufficient permission
```
$ mc admin update local
You are about to upgrade *MinIO Server*, please confirm [y/N]: y

mc: \<ERROR\> Unable to update the server. server update failed with: open /usr/local/bin/.minio.check-perm: permission denied, do not restart the servers yet.
```

This is also mentioned in other issues, e.g. #8226 and a few others.

Clearly this cannot work, since the server, who does a self-update does not have permissions to write to /user/local/bin ; and I'll never give it to it either, for security reasons.

### using a Debian package and using self-update at same time may cause a conflict

It does not make sense to use a Debian package, while the executable is designed to update itself. In my case it did simply not work, at best, it causes a mix of versions.

### move `minio` executable to `/home/minio-user/bin`

We have created a `minio-user` already, so why not move the executable to a place where it can have full access, e.g `/home/minio-user/bin` ?

### change systemd file accordingly

Changed lines:
```
AssertFileIsExecutable=/home/minio-user/bin/minio
WorkingDirectory=/home/minio-user/
ExecStart=/home/minio-user/bin/minio server $MINIO_OPTS $MINIO_VOLUMES
```

### Update works now
```
$ mc admin update local
You are about to upgrade *MinIO Server*, please confirm [y/N]: y
Server `local` updated successfully from 2023-11-20T22:40:07Z to 2023-12-02T10-51-33Z
```

## Thank you
Thanks for creating minio, it's a great piece of software. I enjoy using it, and I am still learning so much every day.

**Comment by harshavardhana** [comment]
No

**Comment by ajdurr** [comment]
can you please explain? Because I have the same issue.

**Comment by edward-b-1** [comment]
Might be helpful https://min.io/docs/minio/linux/operations/install-deploy-manage/upgrade-minio-deployment.html
